webapp/player-project/src/main.ts

```diff
@@ -23,6 +25,19 @@ async function playSessionShadowing(gatewayAccessApi) {
   try {
     const recordingInfo = await gatewayAccessApi.fetchRecordingInfo();
     const fileType = getFileType(recordingInfo);
+    BeforeWebsocketClose((closeEvent) => {
+      if (closeEvent.code !== 1000) {
+        // faild, try to play static recording
+        cleanUpStreamers();
+        playStaticRecording(gatewayAccessApi);
+        return {
+          ...closeEvent,
+          // for avoid extra handling by other listeners, for asciinema-player particularly in this case.
+          // see asciinema-player websocket driver at on socket close handler
+          code: 1000,
```

Comment:
suggestion: Here is an improved wording. Both a stable description and a link are provided since this is an _external_ repository.

```suggestion
          // This prevents extra handling by other listeners, particularly for asciinema-player in this scenario.
          // For more details, see the asciinema-player WebSocket driver’s socket close handler.
          // https://github.com/asciinema/asciinema-player/blob/c09e1d2625450a32e9e76063cdc315fd54ecdd9d/src/driver/websocket.js#L219
          code: 1000,
```